tests/scripts/depends.py

```diff
@@ -331,6 +339,16 @@ def test(self, options):
     'PSA_WANT_ALG_ECB_NO_PADDING' : ['MBEDTLS_NIST_KW_C'],
 }
 
+# Needed for cases when enabling A requires enabling B, but disabling
+# A should not disable B, because B is needed by C, D, E and so on.
+# Entries from this dictionary are added in REVERSE_DEPENDENCIES
+# when running a normal test but not when running the complimentary test.
+SHARED_REVERSE_DEPENDENCIES = {
+        'PSA_WANT_ALG_SHA_224': ['MBEDTLS_PSA_CRYPTO_RNG_HASH'],
+        'PSA_WANT_ALG_SHA_256': ['MBEDTLS_PSA_CRYPTO_RNG_HASH'],
+        'PSA_WANT_ALG_SHA_512': ['MBEDTLS_PSA_CRYPTO_RNG_HASH'],
```

Comment:
Question: why these three hashes and only them? Why not SHA-384 for example?

Reply:
Ok, thanks for clearing that up! But then, why include SHA-224 in this list?

Reply:
I have removed this change and will try to update the other PR's in the pipeline accordingly.

Reply:
Note that I don't think `MBEDTLS_ENTROPY_FORCE_SHA256` ever needed to be on that list, since it's not enabled by `config.py full` anyway. So I don't think we need to particularly preserve it. (Another reason not to preserve it is we are changing things anyway, this is an overhaul of the RNG config system for a new major version, after all.)